components/plurals/src/provider.rs

```diff
@@ -88,3 +89,143 @@ pub(crate) struct ErasedPluralRulesV1Marker;
+/// </div>
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Ord, PartialOrd)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_plurals::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[zerovec::make_ule(RawPluralCategoryULE)]
+#[repr(u8)]
+pub enum RawPluralCategory {
+    /// CLDR "other" plural category.
+    Other = 0,
+    /// CLDR "zero" plural category.
+    Zero = 1,
+    /// CLDR "one" plural category.
+    One = 2,
+    /// CLDR "two" plural category.
+    Two = 3,
+    /// CLDR "few" plural category.
+    Few = 4,
+    /// CLDR "many" plural category.
+    Many = 5,
+}
+
+impl From<PluralCategory> for RawPluralCategory {
+    fn from(value: PluralCategory) -> Self {
+        match value {
+            PluralCategory::Zero => RawPluralCategory::Zero,
+            PluralCategory::One => RawPluralCategory::One,
+            PluralCategory::Two => RawPluralCategory::Two,
+            PluralCategory::Few => RawPluralCategory::Few,
+            PluralCategory::Many => RawPluralCategory::Many,
+            PluralCategory::Other => RawPluralCategory::Other,
+        }
+    }
+}
+
+/// An `u8` that is expected to be a plural range, but does not enforce this invariant.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Ord, PartialOrd)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_plurals::provider),
+)]
+#[zerovec::make_ule(UnvalidatedPluralRangeULE)]
+pub struct UnvalidatedPluralRange(pub u8);
+
+impl UnvalidatedPluralRange {
+    /// Creates a new `UnvalidatedPluralRange` from a category range.
+    pub fn from_range(start: RawPluralCategory, end: RawPluralCategory) -> Self {
+        let start = start as u8;
+        let end = end as u8;
+
+        debug_assert!(start < 16);
+        debug_assert!(end < 16);
+
+        let range = (start << 4) | end;
+
+        Self(range)
+    }
+}
+
+#[cfg(feature = "datagen")]
+impl serde::Serialize for UnvalidatedPluralRange {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use serde::ser::Error;
+        if serializer.is_human_readable() {
+            let start = RawPluralCategory::new_from_u8(self.0 >> 4)
+                .ok_or_else(|| S::Error::custom("invalid tag in UnvalidatedPluralRange"))?;
+            let end = RawPluralCategory::new_from_u8(self.0 & 0x0F)
+                .ok_or_else(|| S::Error::custom("invalid tag in UnvalidatedPluralRange"))?;
+            (start, end).serialize(serializer)
```

Comment:
Thought: if we wanted to make this human-readable we could serialize to a separated string like "One/Other", but I think what you have here is good

Reply:
Should be a pretty quick fix. Maybe something like `One-Other` to use a range-like syntax?

Reply:
Yeah I like that!

(This is a totally optional suggestion; this is mergeable as-is)